packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1568,6 +1570,9 @@ export class KeyringController extends BaseController<
             selector.type,
             options.createWithData,
           )) as SelectedKeyring;
+
+          // This is a new keyring, so we force the vault update in that case.
+          forceUpdate = true;
```

Comment:
As the `withKeyring` overload that creates a new keyring is deprecated, thoughts on removing it completely so that we don't need to manage this case?

Reply:
I think that it could be used in a couple of places, but I don't see blockers from using `addNewKeyring` explicitly

Reply:
As discussed internally, if we don't support this anymore, we probably wants to remove the deprecated method altogether, given that, without the `forceUpdate` logic, the `createIfMissing` option won't work anymore, so I'd just remove it entirely (but we could do that in another PR and keep the force update logic for now)

Reply:
With the new `#getUpdatableState` approach, I was able to get rid of `forceUpdate` entirely! 🎉